### PR TITLE
Updated Core Version 

### PR DESCRIPTION
**Pull request by alberto-ha**
Updated Core Version (web3j:3.4.0, solcJ:0.4.23)
Upgraded Test Smartcontracts (constructor)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,7 +16,9 @@
         <maven.compiler.target>1.8</maven.compiler.target>
         <maven.version>3.5.0</maven.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <web3j.version>3.3.1</web3j.version>
+        <web3j.version>3.4.0</web3j.version>
+        <org.ethereum.solcJ>0.4.23</org.ethereum.solcJ>
```
Comment (h2mch) [inline pom.xml]: the unix version inside this library version is broken

Comment (alberto-ha) [reply pom.xml]: I do not understand when you say with "unix version" and for what library.

Comment (h2mch) [reply pom.xml]: the tests on travis are failing, because of the library "ethereum solcJ". In the version 0.4.23 the solc compiler for unix is broken.

**Comment by h2mch** [comment]
already updated

**Pull request closed without merging**